### ISSUE TITLE
Add an autofill API for system-wide autofill

### ISSUE THREAD

**Issue by nerocui** [microsoft/ProjectReunion#440]
# Proposal: Add an autofill API so apps like Microsoft Authenticator can use it

## Summary
Add APIs/standards for a system-wide autofill service in Windows. App developers like Microsoft Authenticator or LastPass can use it to implement their autofill features on Windows. Windows users should be able to select their preferred autofill provider from the System settings. App developers should be able to adopt this autofill feature in the input fields in their apps with minimal effort (like tag annotation on the web), and they should also be able to extend it should they so require.

## Rationale
* Modern OSes all have this feature. i.e. Android, iOS, macOS
* Apps like LastPass is also available on Windows, but it's not managed by the OS. What happens if the user installs more than one of these apps?
* The new password vault/autofill feature on Microsoft Authenticator is great on Android and iOS, but on Windows, it had to fall back to the browser because there's no such API exist on the OS level. This makes the authentication experience better on the web than native apps. Native apps, such as win32 or UWP, should have feature parity in this regard.

**Comment by KibSquib48** [comment]
Agreed, every other modern system, even linux to some extent, has something like this

**Comment by BenJKuhn** [comment]
This should probably be an OS feature, rather than a reunion feature, since it requires a level access that probably runs deeper than what Project Reunion would provide. @feisu, can you take point on transferring this to an OS ask?
